### ISSUE TITLE
Filter should match <datalist> behavior

### ISSUE THREAD

**Issue by tomayac** [stringyland/custom-select#3]
Rather than just filtering for [`startsWith`](https://github.com/stringyland/custom-select/blob/01e33e27c7c1f53c3ae877a2c4b04b1a26cb7638/index.js#L152), this component should probably match `<datalist>`'s ["contains" filtering behavior](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist#Examples) (type "fox" in the input field in the example).

**Comment by stringyland** [comment]
Good idea - will do some usability testing to see what non-tech-savvy folks expect and update as needed.

**Comment by ollicle** [comment]
I noticed this missing too, although it has potential to bring with it a lot of complexity:
- When the match is not as obvious (as the first letters) the usual strategy to clarify the match by emphasising (e.g. **bold**ing) the matched value in the text of the displayed options.
- Another additional complication _contains_ may bring is the need to reorder the options, e.g. it may make sense to rank full word matches higher than partial.

It would be valuable in assessing these usability and feature questions to detail some specific use cases where custom-select is particularly useful.
